libraries/libfc/src/crypto/bls_private_key.cpp

```diff
@@ -25,67 +25,51 @@ namespace fc::crypto::blslib {
       return bls_private_key(v);
    }
 
-   template<typename Data>
-   Data from_wif( const string& wif_key )
-   {
-   /*   auto wif_bytes = from_base58(wif_key);
-      FC_ASSERT(wif_bytes.size() >= 5);
-      auto key_bytes = vector<char>(wif_bytes.begin() + 1, wif_bytes.end() - 4);
-      fc::sha256 check = fc::sha256::hash(wif_bytes.data(), wif_bytes.size() - 4);
-      fc::sha256 check2 = fc::sha256::hash(check);
+   static std::array<uint64_t, 4> priv_parse_base58(const string& base58str)
+   {  
 
-      FC_ASSERT(memcmp( (char*)&check, wif_bytes.data() + wif_bytes.size() - 4, 4 ) == 0 ||
-                memcmp( (char*)&check2, wif_bytes.data() + wif_bytes.size() - 4, 4 ) == 0 );
+      const auto pivot = base58str.find('_');
+      FC_ASSERT(pivot != std::string::npos, "No delimiter in string, cannot determine data type: ${str}", ("str", base58str));
 
-      return Data(fc::variant(key_bytes).as<typename Data::data_type>());*/
-   }
+      const auto base_prefix_str = base58str.substr(0, 3); //pvt
+      FC_ASSERT(config::bls_private_key_base_prefix == base_prefix_str, "BLS Private Key has invalid base prefix: ${str}", ("str", base58str)("base_prefix_str", base_prefix_str));
+      
+      const auto prefix_str = base58str.substr(pivot + 1, 3); //bls
+      FC_ASSERT(config::bls_private_key_prefix == prefix_str, "BLS Private Key has invalid prefix: ${str}", ("str", base58str)("prefix_str", prefix_str));
+
+      auto data_str = base58str.substr(8);
+
+      std::array<uint64_t, 4> bytes = fc::crypto::blslib::serialize_base58<std::array<uint64_t, 4>>(data_str);
+
+      return bytes;

```

Comment:
This is overly complicated and not quite correct. For example the hardcoded values of 3 and 8 could cause issues. What if the string was just `PVT_`?
I suggest that instead of:
```
      constexpr const char* bls_private_key_base_prefix = "PVT";
      constexpr const char* bls_private_key_prefix = "BLS";
```

you just declare:

```
      constexpr const std::string bls_private_key_prefix = "PVT_BLS_";
```

and you check the prefix with something like:

```
      auto res = std::mismatch(config::bls_private_key_base_prefix.begin(), config::bls_private_key_base_prefix.end(),
                               base58str.begin());
      FC_ASSERT(res.first == config::bls_private_key_base_prefix.end(), ...);
```